src/Agent.Plugins/GitSourceProvider.cs

```diff
@@ -269,6 +309,9 @@ public abstract class GitSourceProvider : ISourceProvider
             }
 
             bool exposeCred = StringUtil.ConvertToBoolean(executionContext.GetInput(Pipelines.PipelineConstants.CheckoutTaskInputs.PersistCredentials));
+            
+            // Read 'fetch by commit' value from the environment variable
+            bool fetchByCommit = StringUtil.ConvertToBoolean(System.Environment.GetEnvironmentVariable("AGENT_GIT_FETCHBYCOMMIT"), false);
```

Comment:
make sure also read from context.variables, so customer can either disable the new behavior through definition variable or machine environment variable.

Reply:
Sure. Do we also want to log the value with `executionContext.Debug` as well?

Reply:
i don't think we need, since it's a back door for customer to disable the new behavior. :)